Meta/build-image-raspberry-pi.sh

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+
+set -e
+
+script_path=$(cd -P -- "$(dirname -- "$0")" && pwd -P)
+
+. "${script_path}/shell_include.sh"
+
+DISK_SIZE=$(($(disk_usage "$SERENITY_SOURCE_DIR/Base") + $(disk_usage Root) + 512 + 300))
+
+if [ ! -d "raspberry-pi-firmware" ]; then
```

Comment:
So the assumption is that we never need to update the boot directory? Can we not do a `git pull` (with rebase and another rm for the linux images if necessary) in the existing directory otherwise, so that you’re always up to date and upstream fixes are applied as soon as possible?

Reply:
Wouldn't a `git pull --ff-only` be good enough? I don't think we need to rebase here.

Reply:
No, since you modify the index by deleting the Linux kernel images.

Reply:
`rm`ing them doesn't update the index? That just updates the working tree.

I just confirmed that
```
git clone --filter tree:0 --branch stable --no-checkout https://github.com/raspberrypi/firmware raspberry-pi-firmware
cd raspberry-pi-firmware
git reset --soft a2e586ba98ce68f7d11b1c717ad8329b95dcb3b6
git sparse-checkout set boot/
git checkout
rm boot/kernel*.img
git pull --ff-only
```
works as expected.

Though it looks like `git pull`ing causes unnecessary files to be downloaded?

Reply:
Oh, you mean the checkout of boot is removed through the sparse-checkout command. That should be it. Otherwise the kernel binaries should be tracked files.